docker-gcloud-pubsub-emulator/Dockerfile

```diff
@@ -12,59 +12,63 @@ ARG POETRYPLUGINEXPORT_VERSION=1.9.0
 # renovate: datasource=github-releases depName=eficode/wait-for
 ARG WAITFOR_VERSION=2.2.4
 
-
 FROM curlimages/curl:8.14.1 AS fetcher
 
 ARG WAITFOR_VERSION
 RUN curl -vsSLo /tmp/wait-for "https://github.com/eficode/wait-for/releases/download/v${WAITFOR_VERSION}/wait-for" && \
-    chmod +x /tmp/wait-for
+  chmod +x /tmp/wait-for
+
+
+FROM gcr.io/google.com/cloudsdktool/google-cloud-cli:529.0.0-emulators AS base
 
+ARG TARGETPLATFORM
+ARG NETCATOPENBSD_VERSION
+
+RUN rm -f /etc/apt/apt.conf.d/docker-clean && \
+  echo 'Binary::apt::APT::Keep-Downloaded-Packages "true";' > /etc/apt/apt.conf.d/keep-cache
+RUN --mount=type=cache,sharing=locked,target=/var/cache/apt,id=apt-cache-$TARGETPLATFORM \
+  --mount=type=cache,sharing=locked,target=/var/lib/apt,id=apt-lib-$TARGETPLATFORM \
+  apt-get update -qy && \
+  apt-get install -qy --no-install-recommends \
+  "netcat-openbsd=${NETCATOPENBSD_VERSION}" python3-pip && \
+  gcloud config set disable_usage_reporting true
```

Comment:
This is copied from the previous final layer, with only a change in how caches are mounted. We now have different caches per architecture; otherwise, amd64 and arm64 packages would get mixed up. See [documentation on cache mount](https://docs.docker.com/reference/dockerfile/#run---mounttypecache) for more details.

Reply:
Great info on `sharing=locked`, I had missed that in the docs but should probably go through and update all my images accordingly!